### PR TITLE
Enhancement/4925 survey timeouts

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue:

- #4925

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/googlesitekit/datastore/user/surveys.js
Comment (tofumatt) [inline assets/js/googlesitekit/datastore/user/surveys.js:350]:
```suggestion
	 * Determines whether the survey is timed out or not.
```

File: assets/js/googlesitekit/datastore/user/surveys.js
Comment (tofumatt) [inline assets/js/googlesitekit/datastore/user/surveys.js:206]: Should these be checks for `isTimedOut === false` rather than `! isTimedOut`? Because `isTimedOut` could be `undefined`… do we want this to behave the same way if it's `false` and `undefined`?

```suggestion
			if ( ! isTimedOut && ! isTimingOut ) {
```

**Review by tofumatt — COMMENTED** [review]
This largely looks good, just a few questions and small comments. I think after that this can get merged 👍🏻 

I tested it locally and it worked well 👍🏻

File: assets/js/googlesitekit/datastore/user/surveys.js
Comment (eugene-manuilov) [reply assets/js/googlesitekit/datastore/user/surveys.js:213]: Those variables are always resolved by the time we get them. Added a comment that explains it.

Comment (tofumatt) [reply assets/js/googlesitekit/datastore/user/surveys.js:213]: Ah, awesome, thanks! I think in that case too it's fine to be less explicit with the checks as anyone reading this for prior art will see the comment and know why we're doing a false-y rather than `=== false` check. 👍🏻

**Review by tofumatt — APPROVED** [review]
Awesome, thanks for the comments. Works great when tested locally 🚢

File: assets/js/components/surveys/SurveyViewTrigger.js
Comment (felixarntz) [inline assets/js/components/surveys/SurveyViewTrigger.js:57]: Why is all this needed? It looks like the `triggerSurvey` action already includes these checks and also ensures the API-based bits are resolved. Why duplicate these conditions in here?

I'm not sure why it's needed, and it makes me slightly worried relying on `useEffect` instead of `useMount`, since this really should only happen once.

Comment (felixarntz) [reply assets/js/components/surveys/SurveyViewTrigger.js:57]: @tofumatt I'll leave it to you and @eugene-manuilov to make a decision on this, but my thinking is that the additional code here is somewhat of a duplicate since the action is already resilient enough to "make sure it works", plus the point about `useEffect` theoretically being more error-prone to calling the action multiple times - while I agree in that I cannot see a point where this happens using the current logic, it feels more error-prone than using `useMount`.

If you still feel the current approach is more resilient, we can leave it, but if you agree, please open a follow-up PR to address this.

**Pull request merged**
